.github/workflows/ci.yml

```diff
@@ -0,0 +1,111 @@
+name: CI-Pipeline
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'develop'
+      - 'main'
+      - 'release/*'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+
+      # - uses: pnpm/action-setup@v3
+      #   with:
+      #     run_install: false
+      #     version: 8
+      # - uses: actions/setup-node@v4
+      #   with:
+      #     cache: pnpm
+      #     node-version: 20
+
+      # https://github.com/pnpm/action-setup#use-cache-to-reduce-installation-time
+      - name: Install Node.js
+        uses: actions/setup-node@v4
+        with:
+          # cache: pnpm
+          node-version: 20
+      - name: Install pnpm
+        uses: pnpm/action-setup@v3
+        id: pnpm-install
+        with:
+          version: 8
+          run_install: false
+      - name: Get pnpm store directory
+        id: pnpm-cache
+        shell: bash
+        run: echo "STORE_PATH=$(pnpm store path)" >> $GITHUB_OUTPUT
+      - uses: actions/cache@v4
+        name: Setup pnpm cache
+        with:
+          path: ${{ steps.pnpm-cache.outputs.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store
+
+      - name: Install Root
+        run: npm i --legacy-peer-deps
```

Comment:
Mir ist nicht aufgefallen, dass die Option notwendig war, wenn wir sie brauchen würde ich aber vorschlagen, sie in die `.npmrc` einzutragen. Dann müssen wir sie nicht wiederholen und sie wird auch bei der lokalen Entwicklung angewandt.

Außerdem würde ich hier `npm ci` statt `npm i` vorschlagen: https://docs.npmjs.com/cli/v8/commands/npm-ci